win32ss/gdi/ntgdi/freetype.c

```diff
@@ -4306,28 +4335,22 @@ ftGdiGetTextMetricsW(
 
             IntLockFreeType();
             pOS2 = FT_Get_Sfnt_Table(Face, ft_sfnt_os2);
-            if (NULL == pOS2)
-            {
-                DPRINT1("Can't find OS/2 table - not TT font?\n");
-                Status = STATUS_INTERNAL_ERROR;
-            }
-
             pHori = FT_Get_Sfnt_Table(Face, ft_sfnt_hhea);
-            if (NULL == pHori)
-            {
-                DPRINT1("Can't find HHEA table - not TT font?\n");
-                Status = STATUS_INTERNAL_ERROR;
-            }
-
             Error = FT_Get_WinFNT_Header(Face, &Win);
 
-            if (NT_SUCCESS(Status))
+            if (NT_SUCCESS(Status) || !Error)
             {
                 FillTM(&ptmwi->TextMetric, FontGDI, pOS2, pHori, !Error ? &Win : 0);
 
                 /* FIXME: Fill Diff member */
                 RtlZeroMemory(&ptmwi->Diff, sizeof(ptmwi->Diff));
             }
+            else
+            {
+                DPRINT1("Can't find HHEA table - not TT font?\n");
```

Comment:
It would be better to keep something like in the original code, where we can know precisely what's missing in the font. Now we loose this info.
EDIT: Your changes in IntGetOutlineTextMetrics() appear to do a better job regarding these DPRINT1s.